ci/actions-templates/freebsd-builds-template.yaml

```
@@ -0,0 +1,73 @@
+# This is ci/actions-templates/freebsd-builds-template.yaml
+# Do not edit this file in .github/workflows
+
+name: FreeBSD (PR) # skip-master skip-stable
+name: FreeBSD (master) # skip-pr skip-stable
+name: FreeBSD (stable) # skip-master skip-pr
+
+on:
+  pull_request: # skip-master skip-stable
+    branches:   # skip-master skip-stable
+      - "*"     # skip-master skip-stable
+  push:         # skip-pr
+    branches:   # skip-pr
+      - master  # skip-pr skip-stable
+      - stable  # skip-pr skip-master
+  schedule: # skip-pr skip-stable
+    - cron: "30 0 * * 1" # Every Monday at half past midnight UTC skip-pr skip-stable
+
+jobs:
+  build:
+    name: Build
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # v2 defaults to a shallow checkout, but we need at least to the previous tag
+          fetch-depth: 0
+      - uses: vmactions/freebsd-vm@v0.1.4
+        id: test
+        with:
+          prepare: pkg install -y git
+          usesh: true
+          run: |
+            #
+            echo "========="
+            echo "Acquire tags for the repo"
+            git fetch --no-tags --prune --depth=1 origin +refs/tags/*:refs/tags/*
+            #
+            echo "========="
+            echo "Display the current git status"
+            git status
+            git describe
+            #
+            echo "========="
+            echo "Prep cargo dirs"
+            mkdir -p ~/.cargo/{registry,git}
+            #
+            echo "========="
+            echo "Set environment variables appropriately for the build"
+            echo "$HOME/.cargo/bin" >> $GITHUB_PATH
+            #
+            echo "========="
+            echo "Install Rustup using ./rustup-init.sh"
+            sh rustup-init.sh --default-toolchain=none --profile=minimal -y
+            # source doesn't exist in this sh so instead it's "."
+            . $HOME/.cargo/env
+            #
+            echo "========="
+            echo "Ensure Beta is up to date"
+            if rustc +beta -vV >/dev/null 2>/dev/null; then
+              rustup toolchain uninstall beta
+            fi
+            rustup toolchain install --profile=minimal beta
+            rustup default beta
+            #
+            echo "========="
+            echo "Ensure we have the components we need"
+            rustup component add rustfmt
+            rustup component add clippy
+            #
+            echo "========="
+            echo "Run the freebsd check"
+            sh ci/raw_init.sh
```

This isn't the ideal script to be running here, you probably meant to use:
```suggestion
            bash ci/run.bash
```

But you'll also need to have set up some environment variables first,

You'll want `TARGET` to be set, `SKIP_TESTS` to be unset/empty, `LIBZ_SYS_STATIC=1` at minimum.

---

You know, I said *bash* for a reason :D

---

Yes but bash wasn't found. Need to install it I guess...